src/compose.ts

```diff
@@ -67,7 +68,24 @@ services:
  *
  * @param c The input composition as a plain JS object
  */
-export function normalize(o: any): Composition {
+export function normalize(o: any): Composition;
+export async function normalize(
+	o: any,
+	fileResolverCb: (path: string) => Promise<ReadStream>,
+): Promise<Composition>;
+export function normalize(
+	o: any,
+	fileResolverCb?: (path: string) => Promise<ReadStream>,
+): Composition | Promise<Composition> {
+	if (fileResolverCb === undefined) {
+		return _normalize(o);
+	} else {
+		const composition = _normalize(o);
+		return expandContext(composition, fileResolverCb);
+	}
+}
+
+function _normalize(o: any): Composition {
```

Comment:
nit:  I'm not a fan of an underscore distinguishing two functions.  I'd prefer a descriptive name that makes it plain what the method will do.

Reply:
Applied: normalizeObjecToComposition